crates/ruff_python_parser/src/parser.rs

```diff
@@ -822,6 +822,10 @@ type X \
     [T] = T
 type X[T] \
     = T
+
+# compound statements
+type X = int; type X = str; type X = type
+class X: type X = int
```

Comment:
I'm confused that you call these compound statements here but simple statements everywhere else.